website/content/docs/ecs/architecture.mdx

```diff
@@ -9,37 +9,51 @@ description: >-
       is called _data plane_ communication.
+1. **ACL Controller:** Automatically provisions Consul ACL tokens for Consul clients and service mesh services
+   in an ECS Cluster.
 
 For more information about how Consul works in general, see Consul's [Architecture Overview](/docs/architecture).
 
-In addition to the long-running Consul Client and Sidecar Proxy containers, there
-are also two initialization containers that run:
-
-1. `discover-servers`: This container runs at startup and uses the AWS API to determine the IP address of the Consul server task.
-1. `mesh-init`: This container runs at startup and sets up initial configuration for Consul and Envoy.
+In addition to the long-running Consul Client and Sidecar Proxy containers, the `mesh-init` container runs
+at startup and sets up initial configuration for Consul and Envoy.
 
 ### Task Startup
 
 This diagram shows the timeline of a task starting up and all its containers:
 
 ![Task Startup Timeline](/img/ecs-task-startup.png)
 
-- **T0:** ECS starts the task. The `discover-servers` container starts looking for the Consul server task’s IP.
-  It waits for the Consul server task to be running on ECS, looks up its IP and then writes the address to a file.
-  Then the container exits.
-- **T1:** Both the `consul-client` and `mesh-init` containers start:
-  - `consul-client` starts up and uses the server IP to join the cluster.
+- **T0:** ECS starts the task. The `consul-client` and `mesh-init` containers start:
+  - `consul-client` uses the `retry-join` option to join the Consul cluster
   - `mesh-init` registers the service for this task and its sidecar proxy into Consul. It runs `consul connect envoy -bootstrap` to generate Envoy’s bootstrap JSON file and write it to a shared volume. After registration and bootstrapping, `mesh-init` exits.
-- **T2:** The `sidecar-proxy` container starts. It runs Envoy by executing `envoy -c <path-to-bootstrap-json>`.
-- **T3:** The `sidecar-proxy` container is marked as healthy by ECS. It uses a health check that detects if its public listener port is open. At this time, the user’s application containers are started since all the Consul machinery is ready to service requests.
-- **T4:** Consul marks the service as healthy by running the health checks specified in the task Terraform. The service will now receive traffic. At this time the only running containers are `consul-client`, `sidecar-proxy` and the user’s application container(s).
+- **T1:** The `sidecar-proxy` container starts. It runs Envoy by executing `envoy -c <path-to-bootstrap-json>`.
+- **T2:** The `sidecar-proxy` container is marked as healthy by ECS. It uses a health check that detects if its public listener port is open. At this time, the user’s application containers are started since all the Consul machinery is ready to service requests. The only running containers are `consul-client`, `sidecar-proxy` and the user’s application container(s).
```

Comment:
```suggestion
- **T2:** The `sidecar-proxy` container is marked as healthy by ECS. It uses a health check that detects if its public listener port is open. At this time, your application containers are started since all Consul machinery is ready to service requests. The only running containers are `consul-client`, `sidecar-proxy`, and your application container(s).
```
I believe the style guide wants us to address the reader directly and avoid 'the user'. It helps maintain an active voice. But this guidance only applies to the Consul practitioner, so please ignore this suggestion if we are referring to an end user other than the Consul user.

Reply:
Ah. Hmm. I'm not too familiar actually. It is an architecture description, rather than guide telling a reader what to do if that matters.

I do think the second person there helps clarify what we set up for you, versus what you need to bring (application containers)